src/libcephfs.cc

```diff
@@ -2420,3 +2420,11 @@ extern "C" void ceph_free_snap_info_buffer(struct snap_info *snap_info) {
   }
   free(snap_info->snap_metadata);
 }
+
+extern "C" int ceph_ioctl(struct ceph_mount_info *cmount, int fd, int command, int* file_attr_out)
```

Comment:
I'm not sure we want to expose ioctls at the API level. I've been creating helper functions that call the defined client functions instead.

See `ceph_add_fscrypt_key`, `ceph_remove_fscrypt_key`, `ceph_get_fscrypt_key_status` etc.

If using the example from other comment, having it call `get_inode_flags` would follow that pattern.

Reply:
I wonder if the requirment is to query i_flag status vis IOCTL, we should implement the interface? Otherwise, why should we support FS_IOC_GETFLAGS is there is no IOCTL support in the interface?